buildSrc/src/main/kotlin/plugin/LibraryPlugin.kt

```diff
@@ -0,0 +1,56 @@
+package plugin
+
+import com.android.build.gradle.LibraryExtension
+import org.gradle.api.JavaVersion
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.dependencies
+
+class LibraryPlugin : Plugin<Project> {
```

Comment:
You need to specify if this is an android library or a java library. This looks like an android library, so maybe name it accordingly.